code/game/machinery/doors/firedoor.dm

```diff
@@ -154,12 +154,18 @@
 	. = TRUE
 	if(!I.tool_use_check(user, 0))
 		return
-	WELDER_ATTEMPT_WELD_MESSAGE
+	if(welded)
+		WELDER_ATTEMPT_UNWELD_MESSAGE
+	else
+		WELDER_ATTEMPT_WELD_MESSAGE
 	if(!I.use_tool(src, user, 40, volume = I.tool_volume))
 		return
 	if(!density) //In case someone opens it while it's getting welded
 		return
-	WELDER_WELD_SUCCESS_MESSAGE
+	if(welded)
+		WELDER_UNWELD_SUCCESS_MESSAGE
+	else
+		WELDER_WELD_SUCCESS_MESSAGE
```

Comment:
```suggestion
	if(welded)
		WELDER_WELD_SUCCESS_MESSAGE
	else
		WELDER_UNWELD_SUCCESS_MESSAGE
```

Reply:
`welded = !welded` should be moved before the success messages for it to make sense.

Reply:
^ that is what I was trying to say, I thought it was above anyway